### PR TITLE
Issue 356  nonuniform partition

### PR DESCRIPTION
**Pull request by adler-j**
Gives a better syntax for nonuniform partitions, old syntax still works.

### CODE REVIEW

File: odl/discr/partition.py [hunk 1]
```diff
@@ -928,6 +929,133 @@ def uniform_partition(min_pt=None, max_pt=None, shape=None, cell_sides=None,
...
+    Examples
+    --------
+    With uniformly spaced points the result is the same as a uniform partition:
+
+    >>> nonuniform_partition([0, 1, 2, 3])
+    uniform_partition(-0.5, 3.5, 4)
+    >>> nonuniform_partition([0, 1, 2, 3], [1, 2])
+    uniform_partition([-0.5, 0.5], [3.5, 2.5], [4, 2])
+
+    Can also perform nonuniform partitions. Note that the containing interval
```
Comment (kohr-h) [inline odl/discr/partition.py]: What can?

File: odl/discr/partition.py [hunk 2]
```diff
@@ -928,6 +929,133 @@ def uniform_partition(min_pt=None, max_pt=None, shape=None, cell_sides=None,
...
+    is calculated by assuming that the points are in the middle of the sub
+    intervals:
```
Comment (kohr-h) [inline odl/discr/partition.py]: Subintervals or sub-intervals

**Comment by kohr-h** [comment]
I like the interface.

Seems we're getting more and more code duplication, mostly in input checks and conversions. At some point, a refactor would be good.

Only minor things, rest is good.

**Comment by kohr-h** [comment]
Addition: use the normalization helpers (see fe6df5e4d06ed368d6e4d969cc419a904ad5b111).

**Comment by adler-j** [comment]
> Addition: use the normalization helpers (see fe6df5e).

Thats magic :O

**Comment by adler-j** [comment]
Merge after CI

**Pull request merged**